web/struct_tree_layer_builder.js

```diff
@@ -156,6 +159,40 @@ class StructTreeLayerBuilder {
     }
   }
 
+  #addImageInTextLayer(node, element) {
+    const { alt, bbox, children } = node;
+    const child = children?.[0];
+    if (!this.#rawDims || !alt || !bbox || child?.type !== "content") {
+      return false;
+    }
+
+    const { id } = child;
+    if (!id) {
+      return false;
+    }
+    const span = document.getElementById(id);
+    if (!span) {
+      return false;
+    }
```

Comment:
I've just tested this patch locally, and unfortunately it seems that there's an *intermittent* bug which may cause this lookup to fail.

The reason is that `StructTreeLayerBuilder.render` was recently made async, in PR #18658, which means that structTree-rendering can now be triggered *before* the textLayer is rendered (i.e. through the annotationLayer) and consequently this lookup can fail.

Off the top of my head I don't know how to fix this nicely (since I only just spotted this), without having to unnecessarily delay things in the viewer.

A simple way to reproduce this problem 100 percent of the time is the following diff:
```diff
diff --git a/web/pdf_page_view.js b/web/pdf_page_view.js
index cafa6b40b..5c0d49aa5 100644
--- a/web/pdf_page_view.js
+++ b/web/pdf_page_view.js
@@ -1072,6 +1072,7 @@ class PDFPageView {
             pdfPage,
             viewport.rawDims
           );
+          this.structTreeLayer.render();
         }

         this.#renderTextLayer();
```